metaflow/flowspec.py

```diff
@@ -811,6 +850,92 @@ def next(self, *dsts: Callable[..., None], **kwargs) -> None:
             )
             raise InvalidNextException(msg)
 
+        # check: switch case using condition
+        if condition is not None:
+            if len(dsts) != 1 or not isinstance(dsts[0], dict):
+                msg = (
+                    "Step *{step}* has an invalid self.next() transition. "
+                    "When using 'condition', provide exactly one dictionary argument "
+                    "mapping condition values to step methods.".format(step=step)
+                )
+                raise InvalidNextException(msg)
+
+            if not isinstance(condition, basestring):
+                msg = (
+                    "Step *{step}* has an invalid self.next() transition. "
+                    "The argument to 'condition' must be a string.".format(step=step)
+                )
+                raise InvalidNextException(msg)
+
+            if foreach is not None or num_parallel is not None:
+                msg = (
+                    "Step *{step}* has an invalid self.next() transition. "
+                    "Switch statements cannot be combined with foreach or num_parallel.".format(
+                        step=step
+                    )
+                )
+                raise InvalidNextException(msg)
+
+            switch_cases = dsts[0]
+            if not switch_cases:
+                msg = (
+                    "Step *{step}* has an invalid self.next() transition. "
+                    "Switch statement cannot have an empty dictionary.".format(
+                        step=step
+                    )
+                )
+                raise InvalidNextException(msg)
+
+            # Validate that condition variable exists
+            try:
+                getattr(self, condition)
+            except AttributeError:
+                msg = (
+                    "Condition variable *self.{var}* in step *{step}* "
+                    "does not exist. Make sure you set self.{var} in this step.".format(
+                        step=step, var=condition
+                    )
+                )
+                raise InvalidNextException(msg)
+
+            # Validate that all switch case values are step methods
+            funcs = []
+            for case_value, step_method in switch_cases.items():
```

Comment:
I am not sure we need to validate this pro-actively here. Can we just not check that the selected one is a step? It feels a bit wasteful to do this. Granted, we won't catch as many errors but we would still catch the ones we care about.